auto-update/Control/AutoUpdate.hs

```diff
@@ -1,21 +1,33 @@
--- | A common problem is the desire to have an action run at a scheduled
--- interval, but only if it is needed. For example, instead of having
--- every web request result in a new @getCurrentTime@ call, we'd like to
--- have a single worker thread run every second, updating an @IORef@.
--- However, if the request frequency is less than once per second, this is
--- a pessimization, and worse, kills idle GC.
+-- | In a multithreaded environment, running actions on a regularly scheduled background thread can dramatically improve performance. 
+-- For example, web servers need to return the current time with each HTTP response. 
+-- For a high-volume server, it's much faster for a dedicated thread to run every second, and write the current time to a shared 'IORef', than it is for each request to make its own call to 'getCurrentTime'.
 --
--- This library allows you to define actions which will either be
--- performed by a dedicated thread or, in times of low volume, will be
--- executed by the calling thread.
+-- But for a low-volume server, whose request frequency is less than once per second, that approach will result in /more/ calls to 'getCurrentTime' than necessary, and worse, kills idle GC.
+--
+-- This library solves that problem by allowing you to define actions which will either be performed by a dedicated thread, or, in times of low volume, will be executed by the calling thread.
```

Comment:
So I'm not super confident this new description is better, but I thought the old one was tricky to understand if you weren't familiar with the problem space:

* Reader needs to know why web requests need the current time
* Reader needs to be familiar with concept of 1 thread writing to a shared value, and others reading it

New version:

* Clarifies that auto-updated is meant for multithreaded programs
* Explains the requirement to return the time with each HTTP response (it should probably be more explicit and say HTTP response headers)
* Is more verbose :(

(Again, not certain the new version is better)